cmake/dependencies.cmake

```diff
@@ -109,50 +73,24 @@ if (NOT LIBAPPIMAGE_SHARED_ONLY)
             LIBRARIES "<SOURCE_DIR>/.libs/libsquashfuse.a;<SOURCE_DIR>/.libs/libsquashfuse_ll.a;<SOURCE_DIR>/.libs/libfuseprivate.a"
             INCLUDE_DIRS "<SOURCE_DIR>"
         )
-    else()
-        message(STATUS "Using system squashfuse")
 
-        import_pkgconfig_target(TARGET_NAME libsquashfuse PKGCONFIG_TARGET squashfuse)
-    endif()
-
-
-    set(USE_SYSTEM_LIBARCHIVE OFF CACHE BOOL "Use system libarchive instead of building our own")
-
-    if(NOT USE_SYSTEM_LIBARCHIVE)
-        message(STATUS "Downloading and building libarchive")
-
-        ExternalProject_Add(
-            libarchive-EXTERNAL
-            URL https://www.libarchive.org/downloads/libarchive-3.3.1.tar.gz
-            URL_HASH SHA512=90702b393b6f0943f42438e277b257af45eee4fa82420431f6a4f5f48bb846f2a72c8ff084dc3ee9c87bdf8b57f4d8dddf7814870fe2604fe86c55d8d744c164
-            CONFIGURE_COMMAND CC=${CC} CXX=${CXX} CFLAGS=${CFLAGS} CPPFLAGS=${CPPFLAGS} LDFLAGS=${LDFLAGS} <SOURCE_DIR>/configure --with-pic --disable-shared --enable-static --disable-bsdtar --disable-bsdcat --disable-bsdcpio --with-zlib --without-bz2lib --without-iconv --without-lz4 --without-lzma --without-lzo2 --without-nettle --without-openssl --without-xml2 --without-expat --prefix=<INSTALL_DIR> --libdir=<INSTALL_DIR>/lib ${EXTRA_CONFIGURE_FLAGS}
-            BUILD_COMMAND ${MAKE}
-            INSTALL_COMMAND ${MAKE} install
-            UPDATE_DISCONNECTED On
-        )
-
-        import_external_project(
-            TARGET_NAME libarchive
-            EXT_PROJECT_NAME libarchive-EXTERNAL
-            LIBRARIES "<INSTALL_DIR>/lib/libarchive.a"
-            INCLUDE_DIRS "<INSTALL_DIR>/include/"
+        target_link_libraries(
+            libsquashfuse
+            INTERFACE liblzma
+            INTERFACE libzstd
+            INTERFACE libzlib
         )
     else()
-        message(STATUS "Using system libarchive")
+        message(STATUS "Using system squashfuse")
 
-        import_find_pkg_target(libarchive LibArchive LibArchive)
+        import_pkgconfig_target(TARGET_NAME libsquashfuse PKGCONFIG_TARGET squashfuse)
     endif()
 
+    message(STATUS "Using system libarchive")
+    import_find_pkg_target(libarchive LibArchive LibArchive)
 
     #### build dependency configuration ####
 
-    # only have to build custom xz when not using system libxz
-    if(TARGET xz-EXTERNAL)
-        if(TARGET squashfuse-EXTERNAL)
-            ExternalProject_Add_StepDependencies(squashfuse-EXTERNAL configure xz-EXTERNAL)
-        endif()
-    endif()
-
     # we need Boost.Algorithm, which does not need to be included explicitly since it's header-only
     # link to Boost::boost to include the header directories
     find_package(Boost 1.53.0 REQUIRED)
```

Comment:
Are desktops such as KDE using Boost? Most likely they are not thrilled about this dependency.